### PR TITLE
New package `@apollo/utils.withrequired`

### PR DESCRIPTION
**Pull request by glasser**

### CODE REVIEW

File: packages/withRequired/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "@apollo/utils.withrequired",
+  "version": "1.0.0",
+  "description": "TypeScript utility type WithRequired",
+  "types": "dist/index.d.ts",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/apollographql/apollo-utils.git",
+    "directory": "withRequired/"
```
Comment (trevor-scheer) [inline packages/withRequired/package.json:9]: (this is still wrong in some places, likely one you modeled after)

```suggestion
    "directory": "packages/withRequired/"
```

File: packages/withRequired/jest.config.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import baseConfig from "../../jest.config.base";
```
Comment (trevor-scheer) [inline packages/withRequired/jest.config.ts:1]: We should probably just not have this file if no tests

**Review by trevor-scheer — CHANGES_REQUESTED** [review]
Expecting top-level tsconfig and package-lock updates. Also a changeset going from v0 -> v1.

File: packages/withRequired/package.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "@apollo/utils.withrequired",
+  "version": "1.0.0",
```
Comment (trevor-scheer) [inline packages/withRequired/package.json:3]: Start at 0 with a changeset `major`
```suggestion
  "version": "0.0.0",
```

File: packages/withRequired/package.json (hunk 1 above)
Comment (trevor-scheer) [reply packages/withRequired/package.json:9]: (fixed all in https://github.com/apollographql/apollo-utils/commit/79c742d2c0d31dc0952de68457216b074f3fed22)

Comment (glasser) [reply packages/withRequired/package.json:9]: done

File: packages/withRequired/jest.config.ts (hunk 2 above)
Comment (glasser) [reply packages/withRequired/jest.config.ts:1]: I added tests instead

**Comment by glasser** [comment]
@trevor-scheer I believe I've taken all your suggestions into account. Since you're out, I'm force merging but will appreciate review when you're back.

**Pull request merged**